<a href="https://colab.research.google.com/github/Sanddeep-K/Capstone_1_EDA_Play_Store_App_Review/blob/main/Sandeep_K_Capstone_1_EDA_Play_Store_App_Review.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Project Title : EDA of Play Store App Reviews**

## <b> Problem Description </b>

The Play Store apps data has enormous potential to drive app-making businesses to success.

New app maker company want to build a successful Mobile App, which should be accepted by huge customer base of Play Store and creatre a positive Brand Image among customers.

This company has given this assignment to me and asked me to have a detailed EDA of given Play Store Data and submit a report to the Management with my Recommendations on below,

1. Most Successful Genres, Categories & Apps on the Play Store
2. Customer Segment wise most preferred App Categories
3. Most reviewed and positively endorsed Genres / Categories & Apps
4. Important factors leads to the success of Mobile App
5. Technology, Specifications & Support requirements
6. Things to be avoided by App Makers
7. Actionable insights for the developers to work on and capture the Android market.

# **Let's Start**

### First of all, let's mount the drive and import libraries those are required throughout the course of our analysis.

In [1]:
# Let's mount Google Drive first

from google.colab import drive
drive.mount('/content/drive/', force_remount=True)

Mounted at /content/drive/


In [3]:
# Let's import required libraries to process data analysis and visualization

import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

## **Loading Datasets**

In [4]:
# Loading the file path and datasets for analysis.
data_path = ('/content/drive/MyDrive/Colab Notebooks/Capstone_Project_1/Play_Store_App_Data/Play Store Data.csv')
reviews_path = ('/content/drive/MyDrive/Colab Notebooks/Capstone_Project_1/Play_Store_App_Data/User Reviews.csv')

psa_data_df = pd.read_csv(data_path)
reviews_df = pd.read_csv(reviews_path)

## **Exploring Dataset - 1**

# Play Store App Data

In [5]:
## Let's have a glance over the number of rows and columns in our dataset

shape = psa_data_df.shape
print("The data set contains {} Records.\nNumber of features are: {}. ".format(shape[0],(shape[1])))

The data set contains 10841 Records.
Number of features are: 13. 


In [6]:
# Let's have a glance on first five rows of our dataset
psa_data_df.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [7]:
## Let's have a glance at last five rows of our dataset
psa_data_df.tail()

App             Category  \
10836                               Sya9a Maroc - FR               FAMILY   
10837               Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                         Parkinson Exercices FR              MEDICAL   
10839                  The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840  iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating             Genres      Last Updated         Current Ver  \
10836       Everyone          Education     July 25, 2017                1.48   
10837       Everyone          Education      July 6, 2018                 1.0   
10838       Everyone            Medical  January 20, 2017                 1.0   
10839     Mature 17+  Books & Reference  January 19, 2015  Varies with device   
10840       Everyone          Lifestyle     July 25, 2018  Varies with device   

              Android Ver  
10836          4.1 and up  
10837          4.1 and up  
10838          2.2 and up  
10839  Varies with device  
10840  Varies with device

## Columns Description : Play Store App Data.

1.  **App :** This column consist of the names of the Mobile Apps.
2.   **Category :** This column consist of the category of the Mobile Apps.
3.   **Rating :** This column consist of the user ratings of the Mobile Apps.
4.   **Reviews :** This column consist of the user reviews of the Mobile Apps.
5.   **Size :** This column consist of the size of the Mobile Apps in KB / MB.
6.   **Install :** This column consist of the number of installations of the Mobile Apps.
7.   **Type :** This column consist of the subscription type (free/paid)of the Mobile Apps.
8.   **Price :** This column consist of the subscription cost of the Mobile Apps.
9.   **Content Rating:** This column consist of the user group category of the Mobile Apps.
10.  **Genres:** This column consist of the Genre type name of the Mobile Apps.
11.  **Last Updated:** This column consist of the Last date of mobile app update performed for the Mobile App.
12.  **Current Ver:** This column consist of the name of the current version of the Mobile App.
13.  **Android Ver:** This column consist of the name of the android version for the Mobile App.

In [8]:
# Let's check : Summary of Critical Information for the dataset.

print(f'The below dataframe gives us feature wise data type, unique values, total count, missing count(NaN).')
print('\n')
def psa_df():
  temp = pd.DataFrame(index =psa_data_df.columns)
  temp['Data_Type'] = psa_data_df.dtypes
  temp['Unique_Count'] = psa_data_df.nunique()
  temp['Count']= psa_data_df.isnull().count()
  temp['Missing_Count'] = psa_data_df.isnull().sum()
  temp['Missing_percentage']= round( temp['Missing_Count'] / temp['Count'] * 100, 2) 
  return temp
psa_df()

The below dataframe gives us feature wise data type, unique values, total count, missing count(NaN).




Data_Type  Unique_Count  Count  Missing_Count  \
App               object          9660  10841              0   
Category          object            34  10841              0   
Rating           float64            40  10841           1474   
Reviews           object          6002  10841              0   
Size              object           462  10841              0   
Installs          object            22  10841              0   
Type              object             3  10841              1   
Price             object            93  10841              0   
Content Rating    object             6  10841              1   
Genres            object           120  10841              0   
Last Updated      object          1378  10841              0   
Current Ver       object          2832  10841              8   
Android Ver       object            33  10841              3   

                Missing_percentage  
App                           0.00  
Category                      0.00  
Rating                       13.60  
Reviews                       0.00  
Size                          0.00  
Installs                      0.00  
Type                          0.01  
Price                         0.00  
Content Rating                0.01  
Genres                        0.00  
Last Updated                  0.00  
Current Ver                   0.07  
Android Ver                   0.03

In [9]:
# Let's check : Duplicated rows by taking only unique app and its last entry.

psa_data_df[psa_data_df.duplicated(subset='App', keep='last')]                  #Keeping last record

App             Category  \
1                                Coloring book moana       ART_AND_DESIGN   
23                            Mcqueen Coloring pages       ART_AND_DESIGN   
36    UNICORN - Color By Number & Pixel Art Coloring       ART_AND_DESIGN   
42                        Textgram - write on photos       ART_AND_DESIGN   
139                             Wattpad 📖 Free Books  BOOKS_AND_REFERENCE   
...                                              ...                  ...   
3825                  Transit: Real-Time Transit App  MAPS_AND_NAVIGATION   
3826         Mapy.cz - Cycling & Hiking offline maps  MAPS_AND_NAVIGATION   
3827                                            Uber  MAPS_AND_NAVIGATION   
3865         Moovit: Bus Time & Train Time Live Info  MAPS_AND_NAVIGATION   
3869                  Maps & GPS Navigation — OsmAnd  MAPS_AND_NAVIGATION   

      Rating  Reviews                Size      Installs  Type Price  \
1        3.9      967                 14M      500,000+  Free     0   
23       NaN       61                7.0M      100,000+  Free     0   
36       4.7     8145                 24M      500,000+  Free     0   
42       4.4   295221  Varies with device   10,000,000+  Free     0   
139      4.6  2914724  Varies with device  100,000,000+  Free     0   
...      ...      ...                 ...           ...   ...   ...   
3825     4.2    43269  Varies with device    5,000,000+  Free     0   
3826     4.5    56443                 43M    1,000,000+  Free     0   
3827     4.2  4928420  Varies with device  100,000,000+  Free     0   
3865     4.4   617477  Varies with device   10,000,000+  Free     0   
3869     4.2    60838  Varies with device    5,000,000+  Free     0   

     Content Rating                           Genres      Last Updated  \
1          Everyone        Art & Design;Pretend Play  January 15, 2018   
23         Everyone  Art & Design;Action & Adventure     March 7, 2018   
36         Everyone          Art & Design;Creativity    August 2, 2018   
42         Everyone                     Art & Design     July 30, 2018   
139            Teen                Books & Reference    August 1, 2018   
...             ...                              ...               ...   
3825       Everyone                Maps & Navigation     July 18, 2018   
3826       Everyone                Maps & Navigation     June 26, 2018   
3827       Everyone                Maps & Navigation    August 6, 2018   
3865       Everyone                Maps & Navigation    August 5, 2018   
3869       Everyone                Maps & Navigation    August 6, 2018   

             Current Ver         Android Ver  
1                  2.0.0        4.0.3 and up  
23                 1.0.0          4.1 and up  
36                 1.0.9          4.4 and up  
42    Varies with device  Varies with device  
139   Varies with device  Varies with device  
...                  ...                 ...  
3825               4.4.7  Varies with device  
3826               6.2.0          4.1 and up  
3827  Varies with device  Varies with device  
3865  Varies with device  Varies with device  
3869  Varies with device  Varies with device  

[1181 rows x 13 columns]

In [ ]:
# As per above table, there are 1181 duplicate records in the given dataset.

### **From the above review on dataset, I have got some observations as below:**

1. Also found 1181 duplicate records in the dataset.
2. Rating has total 1474 null values which contributes 13.60% of the ratings data.
2. Columns names are having white spaces, which may create hurdles during Data Wrangling process
2. Type has 1 null value which contributes 0.01% of the total records.
3. Content_Rating has 1 null value which contributes 0.01% of the total records.
4. Current_Ver has 8 null values which contributes 0.07% of the total records.
5. Android_Ver has 3 null values which contributes 0.03% of the total records.
6. Rating column has more than 10% missing values.
8. Found some data impurities as well, like '+' sign in "Installs" Column, 'M / K' in Size column.

### **Handling Missing Values : Play_Store_Dataset**

1. Creating a copy of original dataset to avoid disturbance in original dataset
2. Finding and replacing missing values appropriately (with Mean / Median)
3. Removing dupications to have correct data for analysis
4. Correcting Column Names for effective data wrangling and analysis

In [10]:
# Creating a copy of  Play Store App data dataset. 

ps_data_df = psa_data_df.copy() 
ps_data_df.head(5)

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

1. Let's handle **1181** duplicate records of App Column

In [11]:
# Dropped duplicated apps from dataframe and keeping only last record.

ps_data_df = ps_data_df.drop_duplicates(subset='App', keep = 'last')

In [12]:
# Let's check New shape of Play Store App data

rows = ps_data_df.shape[0]
columns = ps_data_df.shape[1]
print(f'The number of rows in Play Store dataframe after removing duplicate App names are {rows}.') 
print(f'The number of columns in Play Store dataframe after keeping only required columns for EDA are {columns}.')

The number of rows in Play Store dataframe after removing duplicate App names are 9660.
The number of columns in Play Store dataframe after keeping only required columns for EDA are 13.


2. Let's replace spaces in 'Column' names with 'Underscore' 

In [13]:
ps_data_df.columns = ps_data_df.columns.str.replace(' ', '_')                   # It helps in coding while processing and analysing data